contracts/ERC4626/VenusERC4626Factory.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.25;
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { ERC4626 } from "@openzeppelin/contracts/token/ERC20/extensions/ERC4626.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { VenusERC4626 } from "./VenusERC4626.sol";
+import { IComptroller } from "./Interfaces/IComptroller.sol";
+import { ERC4626Factory } from "./Base/ERC4626Factory.sol";
+import { VToken } from "../VToken.sol";
+import { RewardDistributorInterface } from "./Interfaces/IRewardDistributor.sol"; // Interface for claiming rewards.
+
+/// @title VenusERC4626Factory
+/// @notice Factory for creating VenusERC4626 contracts
+contract VenusERC4626Factory is ERC4626Factory {
+    using SafeERC20 for ERC20;
+
+    /// -----------------------------------------------------------------------
+    /// Immutable params
+    /// -----------------------------------------------------------------------
+
+    /// @notice The XVS token contract
+    ERC20 public immutable XVS;
+
+    /// @notice The address that will receive the liquidity mining rewards (if any)
+    address public immutable REWARD_RECIPIENT;
+
+    /// @notice The Venus Comptroller contract
+    IComptroller public immutable COMPTROLLER;
+
+    /// @notice The reward distributor contract
+    RewardDistributorInterface public immutable REWARD_DISTRIBUTOR;
+
+    /// -----------------------------------------------------------------------
+    /// Storage variables
+    /// -----------------------------------------------------------------------
+
+    /// @notice Maps underlying asset to the corresponding vToken
+    mapping(ERC20 => VToken) public underlyingToVToken;
```

Comment:
This mapping is already stored in the `poolRegistry` associated with the comptroller. We could call something like `comptroller.poolRegistry().getVTokenForAsset(comptroller, asset)`, to get the vToken associated with the asset in that comptroller

This way, we could remove this mapping and the function `updateUnderlyingToVToken`